org.opent2t.sample.hub.superpopular/com.hue.hub/js/thingTranslator.js

```diff
@@ -19,39 +19,170 @@ class Translator {
     }
 
     /**
-     * Get the list of devices discovered through the hub.
+    * Get the hub definition and devices
+    */
+    get(expand, payload) {
+        return this.getPlatforms(expand, payload);
+    }
+
+    /**
+    * Get the list of devices discovered through the hub.
+    */
+    getPlatforms(expand, payload) {
+        if (payload != undefined) {
+            return this._providerSchemaToPlatformSchema(payload, expand);
+        }
+        else {
+            return this._makeRequest(this._devicesPath, 'GET')
+                .then((devices) => {
+                    return this._providerSchemaToPlatformSchema(devices, expand);
+                });
+        }
+    }
+
+    /**
+     * Refreshes the OAuth token for the hub by sending refresh POST to the hue provider
      */
-    getHubResURI() {
-        return this._makeRequest(this._devicesPath, 'GET').then((devices) => {
-
-            var toReturn = {};
-            var filteredDevices = [];
-            for (var hueDeviceID in devices){
-
-                var hueDevice = devices[hueDeviceID];
-                // get the opent2t schema and translator for the hue device
-                var opent2tInfo = this._getOpent2tInfo(hueDevice);
-
-                if (opent2tInfo != undefined) // we support the device                    
-                {
-                    // we only need to return certain properties back
-                    var device = {};
-                    device.name = hueDevice.name;
-
-                    // set the specific device object id to be the id
-                    device.id = hueDeviceID;
-
-                    // set the opent2t info for the hue device
-                    device.openT2T = opent2tInfo;
-                    
-                    filteredDevices.push(device);
+    refreshAuthToken(authInfo) {
```

Comment:
Agreed -- do consider writing helpers/code reuse where possible -- 

---
In reply to: [94527684](https://github.com/openT2T/translators/pull/143#discussion_r94527684) [](ancestors = 94527684,94456369)